Detectors/TPC/workflow/src/CATrackerSpec.cxx

```diff
@@ -73,9 +86,132 @@ DataProcessorSpec getCATrackerSpec(bool processMC, std::vector<int> const& input
       auto& parser = processAttributes->parser;
       auto& tracker = processAttributes->tracker;
       parser = std::make_unique<ClusterGroupParser>();
-      tracker = std::make_unique<o2::tpc::TPCCATracking>();
-      if (tracker->initialize(options.c_str()) != 0) {
-        throw std::invalid_argument("TPCCATracking initialization failed");
+      tracker = std::make_unique<o2::tpc::GPUCATracking>();
+
+      // Prepare initialization of CATracker - we parse the deprecated option string here,
+      // and create the proper configuration objects for compatibility.
+      // This should go away eventually.
+
+      // Default Settings
+      float solenoidBz = -5.00668; // B-field
+      float refX = 83.;            // transport tracks to this x after tracking, >500 for disabling
+      bool continuous = false;     // time frame data v.s. triggered events
+      int nThreads = 1;            // number of threads if we run on the CPU, 1 = default, 0 = auto-detect
+      bool useGPU = false;         // use a GPU for processing, if false uses GPU
+      bool dump = false;           // create memory dump of processed events for standalone runs
+      char gpuType[1024] = "CUDA"; // Type of GPU device, if useGPU is set to true
+      GPUDisplayBackend* display = nullptr;
+
+      const auto grp = o2::parameters::GRPObject::loadFrom("o2sim_grp.root");
+      if (grp) {
+        LOG(INFO) << "Initializing run paramerers from GRP";
+        solenoidBz *= grp->getL3Current() / 30000.;
+        continuous = grp->isDetContinuousReadOut(o2::detectors::DetID::TPC);
+      } else {
+        LOG(ERROR) << "Failed to initialize run parameters from GRP";
+        // should we call fatal here?
+      }
+
+      // Parse the config string
+      const char* opt = options.c_str();
+      if (opt && *opt) {
+        printf("Received options %s\n", opt);
+        const char* optPtr = opt;
+        while (optPtr && *optPtr) {
+          while (*optPtr == ' ') {
+            optPtr++;
+          }
+          const char* nextPtr = strstr(optPtr, " ");
+          const int optLen = nextPtr ? nextPtr - optPtr : strlen(optPtr);
+          if (strncmp(optPtr, "cont", optLen) == 0) {
+            continuous = true;
+            printf("Continuous tracking mode enabled\n");
+          } else if (strncmp(optPtr, "dump", optLen) == 0) {
+            dump = true;
+            printf("Dumping of input events enabled\n");
+          }
+#ifdef BUILD_EVENT_DISPLAY
+          else if (strncmp(optPtr, "display", optLen) == 0) {
```

Comment:
I think the string parsing should be moved out of the spec anyway, just moved it here for now to stay compatible. I can for sure move the `#ifdef` inside the `else if` and throw an error otherwise.